OneCommunityGlobal/HighestGoodNetworkApp
# Eduardo add skeleton loading feature to all components

# Description
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105796041/ddfa19ce-f2e2-4f6f-8e87-e7574f1b6cab)

## Main Change
Created a new `SkeletonLoading` component to replace the old `Loading` component.

## Main Changes Explained
- Implemented the `SkeletonLoading` component, which accepts a `template` prop.
- The component uses switch/case logic to determine which Skeleton Loading mock should be rendered based on the provided template.
- The `template` prop is a string that specifies the name of the component being addressed.
  - Example: `<SkeletonLoading template="TeamMemberTasks" />`, `<SkeletonLoading template="UserProfile" />`, etc.

## How to Test
1. Checkout the branch: `git checkout Eduardo_add_skeleton_loading_feature_to_all_components`
2. Run `npm install` and `npm run start:local` to locally run this PR.
3. Log in as an admin to gain access to all pages.
4. Go to the Dashboard and ensure the page fully loads.
5. Navigate to Reports -> Weekly Summaries Reports and wait for the page to fully load.
6. Visit Other Links -> User Management and wait for the page to fully load.
7. Access the View Profile page and wait for the page to fully load.
8. Verify if the Skeleton Loading mock is being rendered correctly.

## Screenshots or Videos of Changes
### Old Loading Component
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105796041/229e118a-2099-4fd7-af11-20694b23d8a1)

### New SkeletonLoading Component
![Screenshot 2023-06-27 12-30-23](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105796041/2a454f5b-a952-46c5-b043-02d5c3cf6ec0)
![Screenshot 2023-06-27 12-30-30](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105796041/bfa01f2e-8a9a-40b3-8b4f-18092dbae584)
![Screenshot 2023-06-27 12-30-07](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105796041/601908b5-05e0-4f50-a369-a955be3a6d0a)
![Screenshot 2023-06-30 12-00-07](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105796041/24ec85fb-5e93-40c7-9755-92ffb8e8b01b)
![Screenshot 2023-06-30 12-00-28](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105796041/fc507490-b0a6-458f-97f4-9388dae1a932)
